velox/dwio/dwrf/writer/Writer.cpp

```diff
@@ -594,25 +595,29 @@ void Writer::flushInternal(bool close) {
       std::optional<uint32_t> lastRoot;
       std::unordered_map<proto::ColumnStatistics*, proto::ColumnStatistics*>
           statsMap;
-      writer_->writeFileStats([&](uint32_t nodeId) -> proto::ColumnStatistics& {
-        auto entry = footer.add_statistics();
-        if (!encryption || !handler.isEncrypted(nodeId)) {
-          return *entry;
-        }
-
-        auto root = handler.getEncryptionRoot(nodeId);
-        auto groupIndex = handler.getEncryptionGroupIndex(nodeId);
-        auto& group = stats.at(groupIndex);
-        if (!lastRoot || root != lastRoot.value()) {
-          // this is a new root, add to the footer, and use a new slot
-          group.emplace_back();
-          encryption->mutable_encryptiongroups(groupIndex)->add_nodes(root);
-        }
-        lastRoot = root;
-        auto encryptedStats = group.back().add_statistics();
-        statsMap[entry] = encryptedStats;
-        return *encryptedStats;
-      });
+      if (writeColumnStats_) {
+        writer_->writeFileStats(
+            [&](uint32_t nodeId) -> proto::ColumnStatistics& {
+              auto entry = footer.add_statistics();
```

Comment:
Just want to make sure this is really what the writer is doing (code pointer would be nice).  Because the alternative is call `add_statistics` only and early return from the function.

Reply:
That is the current way you implemented it in the tests.  The alternative is just adding the placeholders and not populate it with any statistics.  Just want to make sure which one is actually used in production.

Reply:
I see, yes my approach doesn't write the column statistics at all when the `writeColumnStats` option is set to be false. I think the literal meaning of `writeColumnStats` is to either write the ColumnStats or not write the ColumnStats, and it's different from "writeEmptyColumnStats". I can add a comment in the code.

Reply:
@Yuhta I updated the comment of `writeColumnStats_` in velox/dwio/dwrf/writer/Writer.h as below:
```
  // For tests only. When writeColumnStats_ is set to true, the ColumnStatistics
  // would NOT be written at all. Note that this is different from writing empty
  // ColumnStatistics.
  bool writeColumnStats_;
```

Reply:
I think we need to check what other writers are producing to decide how we produce test data.  Do you have some example code in the Presto Java writer that is doing this?

Reply:
Sure let me take a look tomorrow.

Reply:
@Yuhta The Presto code does not do this when writing DWRF footers, ie. There is no such test in Presto and the writer always write the ColumnStatistics. I guess people haven't met this before in Presto. If you're really concerned about this option, we can add a test that reads an existing file we got from our environment, but the smallest one we had is about 1.5MB. Any other ideas?

Reply:
I am curious where does that file come from since Presto is not writing it?

Reply:
> I am curious where does that file come from since Presto is not writing it?

@Yuhta Hi Jimmy, Both Spark and Presto would write the partitioning column in the data files, it's just that the Parquet files Spark wrote didn't have ColumnStatistics. We used Spark to create the Iceberg tables.

I spent a few hours to create a very small file and add a test reading such file.  But this didn't work at the time being:
1. The tests in TableScanTest and HiveConnectorTest are all based on DWRF and ORC, so we had to create ORC files to test.
2. We found that Velox failed to open such files. I created [Velox can't open the ORC files created by spark-sql](https://github.com/facebookincubator/velox/issues/9645#top) issue

What's your recommendation how we should proceed?

Reply:
There is a `ParquetTableScanTest.cpp` in the code base and you can leverage that an focus on Parquet.

If you are not using ORC in your system we can leave that alone and let someone actually using it to fill in the gap.